src/RpcClient/TransactionManager.cs

```diff
@@ -56,22 +63,28 @@ public TransactionManager(RpcClient rpc)
         /// <returns></returns>
         public TransactionManager MakeTransaction(byte[] script, Signer[] signers = null, TransactionAttribute[] attributes = null)
         {
-            var random = new Random();
-            uint height = rpcClient.GetBlockCount() - 1;
             Tx = new Transaction
             {
                 Version = 0,
-                Nonce = (uint)random.Next(),
+                Nonce = (uint)new Random().Next(),
                 Script = script,
-                Signers = signers,
-                ValidUntilBlock = height + Transaction.MaxValidUntilBlockIncrement,
+                Signers = signers ?? Array.Empty<Signer>(),
                 Attributes = attributes ?? Array.Empty<TransactionAttribute>(),
             };
 
-            RpcInvokeResult result = rpcClient.InvokeScript(script, signers);
-            Tx.SystemFee = long.Parse(result.GasConsumed);
             context = new ContractParametersContext(Tx);
-            signStore = new List<SignItem>();
+
+            QueueWork(async _ =>
```

Comment:
This will not guarantee that the transaction will be with `ValidUntilBlock` and `SystemFee` set at the end of the method, will it? because the Tx it's public and can be used before `Sign`

Reply:
This is true. I was trying to maintain the fluent API, but if this is a concern then MakeTransaction needs to be async. will fix